mantidproject/mantid
# Add default peak function setting to EngDiff UI

**Description of work.**
Added setting for default peak function in the EngDiff UI. This presents the user with a drop-down box of a few relevant functions (the users can still select from the full list in the interactive tool when the fit property browser is open, but the aim of this is to not overwhelm the user with similarly named functions - in particular the back-to-back exponential based ones).

The default peak function has been set to BackToBackExponential for which coefficients have been refined for ENGIN-X (stored in the parameters.xml file). The default peak is set when the fitbrowser is init (and it retrieves this from the UI settigns), this means even if you change the default peak in the config settings after opening the UI, the default peak func is reset to whatever is in the UI settings when you open the interactive menu to add a peak - this is deliberate! This is done so that choice of peak function in other plots etc. outside the UI don't change the defualt peak in the UI.

**To test:**
(1) In the main workbench settings, set the peak function to a `Gaussian` (File > Settings >Fitting > Default peak shape)
(2) Open the EngDiff UI (Interfaces > Diffraction > Engineering Diffraction)
(3) Open the workbench settings and check the default peak shape is now `BackToBackExponential`
(4) Go to the fitting tab, load in some focussed data (to make focussed data follow the instructions in Test 1 here https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html or get the data from PR #29726)
(5) Plot the data  and open the fitting tab and add a peak - it should be a `BackToBackExponential`
(6) Open the UI settings again and change the peak function to  `Lorentzian` - this shouldn't update the peak function in the open UI - you can confirm this by adding another peak, it should still be a `BackToBackExponential`
(7) Close the UI and check that the peak function in the workbench settings is still a `BackToBackExponential`
(8) Open the EngDiff UI, check the UI settings has a default peak of `Lorentzian` (i.e. it remembers what was set in step 6), check the peak function in the main workbench is now a `Lorentzian`.
(9) Change default peak in the main workbench settings to `Gaussian`
(10) Repeat steps 4-5 - this time the peak added should be `Lorentzian` (i.e. the first peak added will always be the default peak in the UI settings)

Fixes #31490





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
